mobile-app/src/components/HeaderOptions.tsx

```diff
@@ -207,9 +216,31 @@ export function editProfileOptions(
       <RoundedButton
         text="Done"
         icon={IconCheck}
-        onPress={() => {
-          router.back();
-        }}
+        onPress={async () => {
+          const cond = await newUsernameIsUnique(userData.username, profileFormData.username)
```

Comment:
move this outside (after line 194) we don't want the onclick to be async

Reply:
also stuff i said in disc
look at https://github.com/owengretzinger/habits-together/pull/106/files